internal/retry/policyprovider.go

```diff
@@ -26,44 +26,52 @@ import (
 	"go.uber.org/yarpc/api/transport"
 )
 
+// PolicyProvider returns a retry policy to use for the request,
+// if no policy is available it should return a default policy.
```

Comment:
The default policy isn't part of the contract. The contract is just "returns the retry policy to use for the given request." It's implied (because you don't mention otherwise) that nil is not a valid policy. Whether there is such a thing as a default is up to the implementation.